vtex-sites/nextjs.store
# chore: make store extendable

## What's the purpose of this pull request?

This PR solves a couple of issues:

The store isn't using the files the CLI will overwrite because they are not being referenced yet or don't exist.

The `postinstall` hook runs whenever we run `bun install` or when we install the package, but we just want the `postinstall` hook to run when we run `bun install` while developing `@faststore/core`, not when we're installing it as a dependency in other project or when we are installing dependencies on the `.faststore` folder.


## How does it work?

To build the store, it creates and uses the customization folder with empty folders and files - implying no customizations. Once the FastStore CLI overwrites these folders and files with actual customizations, they will be used in the store. The store should work just like before.

I added a script to check if the project is inside `node_modules` or `.faststore` folders. If it is, the `postinstall` hook is not shown and a message is shown, warning it was skipped.

## How to test it?

To test the first functionality, just make sure the store was working like before.

The second one is a little bit tricky to test, so *confia*. But if you really want to test, clone this repo inside a  folder called `.faststore` and run `bun install`. You should see the message `Skipped postinstall step for @faststore/core` in your terminal.
